console/collections/src/merkle_tree/tests/batch.rs

```diff
@@ -0,0 +1,675 @@
+    };
+    let expected_leaf1 = match *leaf_index == 1 {
+        true => LeafHash::hash_leaf(leaf_hasher, leaf)?,
+        false => LeafHash::hash_leaf(leaf_hasher, &leaves[1])?,
+    };
+    let expected_leaf2 = match *leaf_index == 2 {
+        true => LeafHash::hash_leaf(leaf_hasher, leaf)?,
+        false => LeafHash::hash_leaf(leaf_hasher, &leaves[2])?,
+    };
+    let expected_leaf3 = match *leaf_index == 3 {
+        true => LeafHash::hash_leaf(leaf_hasher, leaf)?,
+        false => LeafHash::hash_leaf(leaf_hasher, &leaves[3])?,
+    };
+    assert_eq!(expected_leaf0, merkle_tree.tree[3]);
+    assert_eq!(expected_leaf1, merkle_tree.tree[4]);
+    assert_eq!(expected_leaf2, merkle_tree.tree[5]);
+    assert_eq!(expected_leaf3, merkle_tree.tree[6]);
+
+    // Depth 2.
+    let expected_left = PathHash::hash_children(path_hasher, &expected_leaf0, &expected_leaf1)?;
+    let expected_right = PathHash::hash_children(path_hasher, &expected_leaf2, &expected_leaf3)?;
+    assert_eq!(expected_left, merkle_tree.tree[1]);
+    assert_eq!(expected_right, merkle_tree.tree[2]);
+
+    // Depth 1.
+    let expected_left = PathHash::hash_children(path_hasher, &expected_left, &expected_right)?;
+    let expected_right = path_hasher.hash_empty()?;
+    assert_eq!(expected_left, merkle_tree.tree[0]);
+
+    // Depth 0.
+    let expected_root = PathHash::hash_children(path_hasher, &expected_left, &expected_right)?;
+    assert_eq!(expected_root, *merkle_tree.root());
+    Ok(())
+}
+
+#[test]
+fn test_merkle_tree_bhp() -> Result<()> {
+    fn run_test<const DEPTH: u8>(rng: &mut TestRng) -> Result<()> {
+        type LH = BHP1024<CurrentEnvironment>;
+        type PH = BHP512<CurrentEnvironment>;
+
+        let leaf_hasher = LH::setup("AleoMerkleTreeTest0")?;
+        let path_hasher = PH::setup("AleoMerkleTreeTest1")?;
+
+        for i in 0..ITERATIONS {
+            for j in 0..ITERATIONS {
+                // Determine the leaves and additional leaves.
+                let num_leaves = core::cmp::min(2u128.pow(DEPTH as u32), i);
+                let num_updates = core::cmp::min(num_leaves, core::cmp::min(2u128.pow(DEPTH as u32) - num_leaves, j));
+
+                // Check the Merkle tree.
+                check_merkle_tree::<CurrentEnvironment, LH, PH, DEPTH>(
+                    &leaf_hasher,
+                    &path_hasher,
+                    &(0..num_leaves)
+                        .map(|_| Field::<CurrentEnvironment>::rand(rng).to_bits_le())
+                        .collect::<Vec<Vec<bool>>>(),
+                    &(0..num_updates)
+                        .rev()
+                        .map(|i| ((i % num_leaves) as usize, Field::<CurrentEnvironment>::rand(rng).to_bits_le()))
+                        .collect::<Vec<(usize, Vec<bool>)>>(),
+                )?;
+            }
+        }
+        Ok(())
+    }
+
+    let mut rng = TestRng::default();
+
+    // Ensure DEPTH = 0 fails.
+    assert!(run_test::<0>(&mut rng).is_err());
```

Comment:
[`seq_macro`](https://docs.rs/seq-macro/0.3.3/seq_macro/) could be used to reduce some of this boilerplate; it would look more or less like this:
```
seq!(N in 1..=10 {
    assert!(run_test::<N>(&mut rng).is_ok());
});
```
alternatively (or in addition), a macro like this could be used:
```
macro_rules! run_tests {
    ($rng:expr, [$($i:expr),*]) => {
        $( assert!(run_test::<$i>($rng).is_ok()); )*
    };
}

run_tests!(&mut rng, [1, 2, 10]); // just listing a few
```

Reply:
Opting for the second solution.